pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -432,53 +432,45 @@ def copy_trainer_model_properties(self, model):
             m.tpu_local_core_rank = self.tpu_local_core_rank
             m.tpu_global_core_rank = self.tpu_global_core_rank
 
-    def transfer_batch_to_tpu(self, batch):
-        return self.__transfer_data_to_device(batch, device='tpu')
+    def transfer_batch_to_tpu(self, batch: Any, tpu_id: int = None):
+        device = xm.xla_device(tpu_id) if XLA_AVAILABLE else torch.device('cpu')
```

Comment:
if tpu is not available should we really default to cpu silently?

I think we should at least raise a Warning and maybe also check for gpu and prefer this over cpu (if available)

Reply:
It was like this before as far as I can tell. 
Agree a warning would be appropriate here.
Not sure if we can automatically make the decision to run on GPUs. How many would we choose? One? All of them?

Reply:
warning to log, probably no need for runtime warning...?

Reply:
@Borda I'd even throw a runtime warning here. Otherwise user will wonder why training takes so long and maybe won't look into logs

@awaelchli I'd try using the tpu_id since the assumption in that case was, that the user mistakenly confused TPU and GPU

Reply:
sounds reasonable. by runtime warning you mean rank_zero_warn or something else?

Reply:
actually I just saw it's not so easy to switch to GPU when self.use_tpu is already True because the distributed_backend setup was already disabled, so I think this is something we should do at the beginnng of trainer and not here where it is too late. 
Let's just raise an error instead here? 
there was another issue mentioned recently about what happens when user both passes gpus and tpu_ids into Trainer. We can probably combine these two edge cases into a separate PR, what do you think?